torchtune/datasets/_text_completion.py

```diff
@@ -61,6 +74,11 @@ def __getitem__(self, index: int) -> Dict[str, List[int]]:
 
     def _prepare_sample(self, sample: Mapping[str, Any]) -> Dict[str, List[int]]:
         prompt = sample[self._column]
+
+        # Apply the filter function to the text data
+        if self.filter_fn:
```

Comment:
let's apply the filter_fn the same way as in `_sft.py`:
https://github.com/pytorch/torchtune/blob/f95a9f8cccb61e55faacdeb2efdb30a8dd3d33f8/torchtune/datasets/_sft.py#L110
it's consistent with how HF handles filtering and makes the contract for the filter_fn very clear (returning a bool for whether the data sample should be kept or not)